developers/weaviate/manage-data/collections.mdx

```diff
@@ -0,0 +1,469 @@
+        },
+        {
+            "dataType": ["object"],
+            "name": "address",
+            "nestedProperties": [
+                {"dataType": ["text"], "name": "street"},
+                {"dataType": ["text"], "name": "city"}
+            ],
+        }
+    ],
+}
+```
+
+<details>
+  <summary>Additional information</summary>
+
+The following code defines:
+
+- The collection name (`Person`)
+- The vectorizer module (`text2vec-huggingface`)
+- A set of properties (`last_name`, `address`)
+
+    - `last_name` has `text` data type
+    - `address` has `object` data type
+    
+- The `address` property has two nested properties (`street` and `city`)
+
+</details>
+
+## Sample configuration: Generative search
+
+Collection configuration for [generative search](../search/generative.md) on text objects.
+
+```json
+{
+    "class": "Article",
+    "vectorizer": "text2vec-openai",
+    "vectorIndexConfig": {
+        "distance": "cosine",
+    },
+    "moduleConfig": {
+        "generative-openai": {}
+    },
+    "properties": [
+        {
+            "name": "title",
+            "dataType": ["text"]
+        },
+        {
+            "name": "chunk",
+            "dataType": ["text"]
+        },
+        {
+            "name": "chunk_no",
+            "dataType": ["int"]
+        },
+        {
+            "name": "url",
+            "dataType": ["text"],
+            "tokenization": "field",
+            "moduleConfig": {
+                "text2vec-openai": {
+                    "skip": true
+                },
+            }
+        },
+    ],
+}
+
+```
+
+<details>
+  <summary>Additional information</summary>
+
+The collection defines the following:
+
+- The collection name (`Article`)
+- The default vectorizer module (`text2vec-openai`)
+- The generative module (`generative-openai`)
+- A set of properties (`title`, `chunk`, `chunk_no` and `url`)
+- The tokenization option for the `url` property
+- The vectorization option (`skip` vectorization) for the `url` property
+
+</details>
+
+## Sample configuration: Images
+
+Image collection configuration details.
+
+```json
+{
+    "class": "Image",
+    "vectorizer": "img2vec-neural",
+    "vectorIndexConfig": {
+        "distance": "cosine",
+    },
+    "moduleConfig": {
+        "img2vec-neural": {
+            "imageFields": [
+                "image"
+            ]
+        }
+    },
+    "properties": [
+        {
+            "name": "image",
+            "dataType": ["blob"]
+        },
+    ],
+}
+```
+
+<details>
+  <summary>Additional information</summary>
+
+The collection defines the following:
+
+- The collection name (`Image`)
+- The vectorizer module (`img2vec-neural`)
+
+  - The `image` property configures collection to store image data.
+  
+- The vector index distance metric (`cosine`)
+- A set of properties (`image`), with the `image` property set as `blob`.
+
+For image searches, see [Image search](../search/image.md).
+
+</details>
```

Comment:
The sample configuration only provides Python v3 examples.

Can we add examples for Python v4 and TypeScript?

Reply:
Also, can we use code highlighting for the essential parts of the code.
For example, in the example showcasing generative configuration, we should highlight the 3 lines that show that configuration:
```
    "moduleConfig": {
        "generative-openai": {}
    },
```

Reply:
@sebawita I thought json meant those were generic configs, that's why I moved them to a sample section.  I can open another ticket to update them

Reply:
sounds good 👍 
It looks like this got addressed in #1435 😄